### PR TITLE
cidr-fix for vsphere cloud provider

### PR DESCRIPTION
**Pull request by imkin**
This change is specific to vsphere cloud provider.
The vsphere cloud provider would earlier set cidr ranges for cbr0 bridge for each node. This does not work anymore and is probably not recommended. 

let kubelet configure cbr0 bridge.
let kube-control-manager distributed the subnet across the nodes.
create routes for the pod network.

### CODE REVIEW

File: cluster/vsphere/util.sh [hunk 1]
```diff
@@ -306,6 +303,41 @@ function remote-pgrep {
...
+  # identify the subnet assigned to the minion by the kube-controll-manager.
+  KUBE_NODE_BRIDGE_NETWORK=()
+  for (( i=0; i<${#NODE_NAMES[@]}; i++)); do
+     printf " finding network of cbr0 bridge on node  ${NODE_NAMES[$i]}\n"
+     network=$(kube-ssh ${KUBE_NODE_IP_ADDRESSES[$i]} 'sudo ip route show | grep -E "dev cbr0" | cut -d     " " -f1')
+     KUBE_NODE_BRIDGE_NETWORK+=("${network}")
+  done
+
+
+  # make the pods visible to eachother.
```
Comment (davidopp) [inline cluster/vsphere/util.sh]: s/eachother/each other/

but not a big deal, can do in a follow-up PR

File: cluster/vsphere/util.sh [hunk 2]
```diff
@@ -306,6 +303,41 @@ function remote-pgrep {
...
+function setup-pod-routes {
+  # wait till the kubelet setups the bridge.
+  echo "Setting up routes"
+  for (( i=0; i<${#NODE_NAMES[@]}; i++)); do
+     printf "check if cbr0 bridge is ready on ${NODE_NAMES[$i]}\n"
+     kube-check ${KUBE_NODE_IP_ADDRESSES[$i]} 'sudo ifconfig cbr0 | grep -oP "inet addr:\K\S+"'
+  done
+
+
+  # identify the subnet assigned to the minion by the kube-controll-manager.
```
Comment (davidopp) [inline cluster/vsphere/util.sh]: s/minion/node/
s/kube-controll-manager/kubernetes controller manager/

File: cluster/vsphere/util.sh (hunk 1 above)
Comment (davidopp) [inline cluster/vsphere/util.sh]: actually, since I found a couple of other typos, please fix this one too.

File: cluster/vsphere/util.sh [hunk 3]
```diff
@@ -306,6 +303,41 @@ function remote-pgrep {
   done
 }
 
+# identify the pod routes and route them together.
+#
+# Assumptions:
+#  All packages have been installed and kubelet has started running.
+#
+function setup-pod-routes {
+  # wait till the kubelet setups the bridge.
```
Comment (davidopp) [inline cluster/vsphere/util.sh]: s/setups/sets up/

**Comment by imkin** [comment]
@davidopp I have applied your comments :-)

**Comment by davidopp** [comment]
LGTM

**Comment by imkin** [comment]
@davidopp can we mark this for milestone 1.2 [ so that this is part of the release] ?

**Comment by davidopp** [comment]
Done but there is no guarantee it will merge by the cutoff (Friday 11 AM Pacific Time). If it doesn't, you'll need to prepare and submit a cherrypick. Instructions for doing that are here:
https://github.com/kubernetes/kubernetes/blob/ad68688a52f8514772c8e480f02ca677428770e7/docs/devel/cherry-picks.md

**Comment by imkin** [comment]
@dchen1107 looks like a e2e failed ... unstable test?

**Comment by davidopp** [comment]
@k8s-bot e2e test this issue: #18558

**Pull request merged**